wandb/sdk/lib/redirect.py

```diff
@@ -620,13 +688,19 @@ def uninstall(self):
         if not self._installed:
             return
         self._installed = False
+
         self.src_wrapped_stream.flush()
-        time.sleep(1)
-        self._stopped.set()
+        time.sleep(0.1)
         os.dup2(self._orig_src_fd, self.src_fd)
         os.write(self._pipe_write_fd, b"\n")
         os.close(self._pipe_write_fd)
         os.close(self._pipe_read_fd)
+
+        # Joining daemonic thread might hang, so we wait for the queue to empty out instead:
+        while not self._queue.empty():
+            time.sleep(0.1)
```

Comment:
Is there a scenario where the queue never emptys?

Reply:
Hmm.. in theory no. But lets bail after a while just in case?

Reply:
Yeah, I suppose we could do this for 10 seconds and give up with a warning if we ever take that long.  Dropping log output isn't the end of the world.